pkg/rpcclient/actor/waiter.go

```diff
@@ -260,72 +315,20 @@ func (w *EventWaiter) WaitAny(ctx context.Context, vub uint32, hashes ...util.Ui
 				}
 			}
 		}
-		if wsWaitErr == nil || !errors.Is(wsWaitErr, ErrMissedEvent) {
-			close(bRcvr)
-			close(aerRcvr)
-		}
-	}()
-
-	// Execution event preceded the block event, thus wait until the VUB-th block to be sure.
-	since := vub
-	blocksID, err := w.ws.ReceiveBlocks(&neorpc.BlockFilter{Since: &since}, bRcvr)
-	if err != nil {
-		wsWaitErr = fmt.Errorf("failed to subscribe for new blocks: %w", err)
-		return
 	}
-	waitersActive++
-	defer func() {
-		go func() {
-			err = w.ws.Unsubscribe(blocksID)
-			if err != nil {
-				unsubErrs <- fmt.Errorf("failed to unsubscribe from blocks (id: %s): %w", blocksID, err)
-				return
-			}
-			unsubErrs <- nil
-		}()
-	}()
-	for _, h := range hashes {
-		txsID, err := w.ws.ReceiveExecutions(&neorpc.ExecutionFilter{Container: &h}, aerRcvr)
-		if err != nil {
-			wsWaitErr = fmt.Errorf("failed to subscribe for execution results: %w", err)
-			return
-		}
-		waitersActive++
-		defer func() {
-			go func() {
-				err = w.ws.Unsubscribe(txsID)
-				if err != nil {
-					unsubErrs <- fmt.Errorf("failed to unsubscribe from transactions (id: %s): %w", txsID, err)
-					return
-				}
-				unsubErrs <- nil
-			}()
-		}()
+	if wsWaitErr == nil || !errors.Is(wsWaitErr, ErrMissedEvent) {
```

Comment:
Why this condition for closing these channels? They're always initialized and they're drained above.